### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by jhanca-robotecai**
Readme updates.

Draft: more changes to come.

### CODE REVIEW

**Comment by jhanca-robotecai** [comment]
Switching this PR to _ready to review_.

This PR finalizes the documentation in the following files:
- README.md
- docs/interfaces.md
- docs/newFeature.md
- docs/newVendor.md

Besides, it adds new diagrams and placeholders for documentation that will be fixed in separate PRs.

**Comment by jhanca-robotecai** [comment]
Pushed some changes in `vendors.md` and `gettingStarted.md`. The first one will be further improved.

File: README.md [hunk 1]
```diff
@@ -1,59 +1,71 @@
-# o3de-genai-gems
+# O3DE Generative AI Gems collection
 
-`o3de-genai-gems` repo contains the core Gem of the Gen AI O3DE Framework, a sample Gen AI Feature Gem for assisting with O3DE related questions and a collection of Gen AI Vendor Gems.
+The `o3de-genai-gems` repository serves as a hub for the development and distribution of O3DE Gems developed to support Generative AI models within O3DE. In particular, it contains the core Gem of the Generative AI O3DE Framework, a sample Generative AI Feature Gem for assisting with O3DE-related questions and a collection of Generative AI Vendor Gems to support the communication between the AI client and AI server.
+
+## Introduction
 
-## Context
 [Open 3D Engine](https:://o3de.org) - an open-source game & simulation engine. O3DE is extendable through modules called Gems. This repository is a collection of such Gems related to one functionality.
```
[the remark this answers is not in the document]
Comment (jhanca-robotecai) [reply README.md:7]: There was a double-double-dot, fixed

File: README.md [hunk 2]
```diff
@@ -1,59 +1,71 @@
...
+Vendor Gems provide mechanisms to communicate with certain generative AI models running on certain servers. This repository contains two Gems that bundle multiple AI models and service definitions each:
+- `GenAIVendorBundle`: a vendor Gem that implements communication with [Anthropic](https://docs.anthropic.com/en/docs/welcome) service for running `Claude` model and [Ollama toolset](https://github.com/ollama/ollama) that allows getting up and running various language models locally.
+- `GenAIAmazonBedrockVendorBundle Gem`: a vendor Gem that implements communication with [Amazon Bedrock service](https://aws.amazon.com/bedrock/) and that offers a choice of high-performing models from leading AI companies. This gem is dependent on the `GenAIVendorBundle` Gem and allows for communication with the `Claude` model using the `Amazon Bedrock` service.
+
+Additionally, a mocked vendor Gem that can produce predictable responses based on the given input data in JSON format is provided in `GenAIMock`. This Gem is meant for cost-effective and fast tests as well as a sample implementation of a vendor Gem.
```
[the remark this answers is not in the document]
Comment (jhanca-robotecai) [reply README.md:55]: Great to see that humans still outperform AI when creating technical documentation ;)

File: docs/gettingStarted.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Add the registered Gems to your O3DE project:
+```bash
+scripts/o3de.sh enable-gem --all-gems-path <o3de-genai-gems>/Gems --project-name <project name>
+```
+
+For a complete tutorial on project configuration, see [Creating Projects Using the Command Line Interface](https://o3de.org/docs/welcome-guide/create/creating-projects-using-cli/) in the O3DE documentation.
+
+## Setting up a new vendor and creating a AI Assistant chat
+
+This is an example of how to set up a vendor in O3DE. It explains how to create and use the `Claude 3` models with the `Claude HTTP` service. Other vendors can be set up similarly.
```
[the remark this answers is not in the document]
Comment (jhanca-robotecai) [reply docs/gettingStarted.md:21]: I am aware this part is missing, I have this in my backlog. I will create an issue to be sure it is solved before the release.

[the remark this answers is not in the document]
Comment (jhanca-robotecai) [reply docs/gettingStarted.md:21]: https://github.com/RobotecAI/o3de-genai-gems/issues/143

File: docs/newFeature.md [hunk 4]
```diff
@@ -1,48 +1,49 @@
...
 #include <GenAIFramework/Feature/FeatureBase.h>
 
 class MyFeature : public FeatureBase
-    {
-    public:
-        AZ_RTTI(MyFeature, "{00000000-0000-0000-0000-000000000000}", FeatureBase);
-        AZ_CLASS_ALLOCATOR(MyFeature, AZ::SystemAllocator, 0);
+{
+public:
+    AZ_RTTI(MyFeature, "{00000000-0000-0000-0000-000000000000}", FeatureBase);
```
[the remark this answers is not in the document]
Comment (jhanca-robotecai) [reply docs/newFeature.md:20]: The current code sample is correct. This is the following example:
```
AZ_RTTI(<this_class>, <uuid_number>, <base_class>);
```

File: README.md [hunk 5]
```diff
@@ -1,59 +1,71 @@
...
 1. A core framework Gem that is intended for developers to build on. It is a bridge between the user, O3DE and the AI: other Gems will use it to build features powered by Generative AI.
 2. A collection of vendor Gems used to connect to specific AI services or models.
 3. A collection of feature Gems that execute certain functionalities in O3DE based on Gen AI responses, such as level generation/modification or character control.
 
 In other words, a feature Gem prepares the query to the Generative AI (e.g. asks to modify the level's layout given the current layout) based on the user's input and a vendor Gem is responsible for converting this query into a meaningful format and connecting with a given AI model. _GenAIFramework_ Gem connects the two, i.e., it ensures that all vendor Gems are available for feature Gems. Importantly, it also provides the user interface for AI communication. The framework is summarized in the picture below.
 
-![Framework Design](docs/images/GenAIFramework.png)
+![Framework Design](./docs/images/GenAIFramework.png)
```
[the remark this answers is not in the document]
Comment (jhanca-robotecai) [reply README.md:16]: Created the issue: https://github.com/RobotecAI/o3de-genai-gems/issues/144

**Review by PawelLiberadzki — APPROVED** [review]
Looks good!

**Pull request merged**